src/server/game/AI/SmartScripts/SmartScript.h

```diff
@@ -294,23 +296,25 @@ class SmartScript
             }
         }
     }
-    SmartScriptHolder FindLinkedEvent (uint32 link)
+    std::optional<std::reference_wrapper<
+        SmartScriptHolder>> FindLinkedEvent(uint32 link)
     {
         if (!mEvents.empty())
         {
             for (SmartAIEventList::iterator i = mEvents.begin(); i != mEvents.end(); ++i)
             {
                 if (i->event_id == link)
                 {
-                    return (*i);
+                    return std::ref(*i);
                 }
             }
         }
-        SmartScriptHolder s;
-        return s;
+        return std::nullopt;
     }
```

Comment:
Lol, I was wondering why it was implemented that way. 
I thought maybe it was because the object’s lifetime is short and we need to work with copies of the objects. 
But then I noticed that we're not clearing the mEvents vector 😅
So, good catch!